dht_options.go

```diff
@@ -175,6 +191,8 @@ func Mode(m ModeOpt) Option {
 // Validator configures the DHT to use the specified validator.
 //
 // Defaults to a namespaced validator that can only validate public keys.
+// For the default protocol prefix it defaults to a namespaced validator that
+// supports validating both public key and IPNS records
```

Comment:
This is the weird comment we sort of have to leave now if we stick with this approach https://github.com/libp2p/go-libp2p-kad-dht/pull/560/files#r404190615

Reply:
I updated things to make them a little simpler and more compliant, but now this comment isn't 100% accurate because the internals have more subtleties. Thoughts?